models/app.py

```diff
@@ -0,0 +1,119 @@
+import datetime
+import json
+import logging
+import os
+from datetime import datetime
+
+from db_models import Activity
+from flask import Flask, jsonify, request
+from flask_mongoengine import MongoEngine
+from mongoengine.queryset.visitor import Q
+
+DATE_FMT = '%Y-%m-%d %H:%M:%S'
+
+log = logging.getLogger()
+logging.basicConfig(filename="models/mongo.log", filemode='a',
+                    format='%(asctime)s, %(levelname)s %(message)s',
+                    datefmt='%H:%M:%S',
+                    level=logging.DEBUG)
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def index():
+    return jsonify(status=200, message='MongoDB app!', data=[])
+
+
+@app.route('/activity', methods=['POST'])
+def add_activity():
+    data_dict = request.get_json(force=True)
+    if not isinstance(data_dict, dict):
+        return jsonify(status=400, message="Invalid payload data. Data "
+                                           "should be single JSON", data=[])
+    _items = list()
+    for item in data_dict['items']:
+        _new_activity = Activity()
+        _new_activity = _new_activity.from_json(json.dumps(item))
+        try:
+            _new_activity.validate()
+        except Exception as e:
+            log.error(e.to_dict())
+            return jsonify(status=400, message=e.to_dict(), data=[])
+        _items.append(_new_activity)
+    Activity.objects.insert(_items)
+    msg = "Total {} activities added successfully.".format(len(_items))
+    log.info(msg)
+    return jsonify(status=200, message=msg, data=[])
+
+
+@app.route('/activity/<int:ph_no>')
+def get_activity(ph_no):
+    data_dict = request.get_json(force=True)
+    filters = Q(phone_no=ph_no)
+    if 'start_date' in data_dict:
+        try:
+            _start_date = datetime.strptime(data_dict['start_date'], DATE_FMT)
+        except ValueError:
+            _start_date = datetime.strptime(data_dict['start_date'] +
+                                            " 00:00:00", DATE_FMT)
+        finally:
+            filters = filters & Q(recorded_at__gt=_start_date)
+
+    if 'end_date' in data_dict:
```

Comment:
That logic will not work properly.
Eg, 
we have data for `user-A` for the following day:
```
1st Jan, 2nd Jan, 4th Jan, 7th Jan, 9th Jan.
```

Now, if the data between 3rd Jan and 7th Jan is requested then,
This will return data for (`1st Jan, 2nd Jan, 4th Jan, 7th Jan`) instead of just returning data for (`4th Jan, 7th Jan`)